### DISCUSSION TITLE
SemVer condition

### DISCUSSION THREAD

**Discussion by asbjornu** [Mergifyio/mergify#2315] [Ideas]
In Dependabot, it's possible to auto-merge only if the version bump is "in-range" and other semantic version restrictions that don't seem to be possible to reproduce with Mergify.

<img width="522" alt="image" src="https://user-images.githubusercontent.com/12283/112516046-27aa9180-8d97-11eb-835e-81c75fc706f9.png">

Would it be possible to add conditions like this or is this possible somehow already?

**Comment by jd** [comment]
There's one example to do such a thing using regular expression here:
https://docs.mergify.io/examples.html#dependabot

Let me know if that works for you, or if you come up with anything else worth adding to the example list!

**Reply by asbjornu** [reply]
Thanks for the pointer, but I've seen that example and am sorry to say it does not provide the functionality I'm after. What I need is for Mergify to be able to respect the version restrictions defined in the manifest, such as this:

```ruby
gem 'jekyll', '>= 4.0.0', '< 4.2.0'
```

I.e. the gem `jekyll` should be at least version `4.0.0`, but lower than version `4.2.0` (so version `4.1.7` is okay).

I do want Dependabot to submit pull requests that violate this restriction, in order to trigger automatic tests and thus providing feedback on whether the upgrade will work despite the pessimistic expectations codified in the manifest.

I do, however, not want pull requests that violate this version restriction to be automatically merged, since I want to do a manual review of the upgrade and verify that stuff possibly not caught by the tests are, indeed, working.

**Reply by jd** [reply]
Ah. That sounds like a dependabot feature, e.g., they could add a custom label if they violate your restriction on purpose.

Mergify has no plan to learn how to parse Ruby manifest files for now 😅

**Reply by asbjornu** [reply]
Okay, that makes sense. If Dependabot provided such a label, would Mergify be able to be configured to not merge when it's present?

**Reply by sileht** [reply]
Yes, the conditions just need to have something like: `- label!=major-version`
